aptos-move/block-executor/src/executor.rs

```diff
@@ -69,8 +72,11 @@ impl<
     pub fn read(&self, key: &K) -> ReadResult<V> {
         use MVHashMapError::*;
         use MVHashMapOutput::*;
-
         loop {
+            // If the PE halts, read should return immediately.
+            if self.scheduler.done() {
```

Comment:
this check is probably no longer needed since we do check it while holding the lock.